iscript/src/iscript/macvpn.py

```diff
@@ -64,20 +65,133 @@ async def _sign_app(config, sign_config, app, entitlements_url, provisionprofile
     """
     # We mock the task for downloading entitlements and provisioning profiles
     entitlements_path = await download_entitlements_file(config, sign_config, {"payload": {"entitlements-url": entitlements_url}})
+
+    # TODO: Add provisionprofile_dir to scriptworker config
+    # https://github.com/mozilla-releng/scriptworker/blob/master/src/scriptworker/constants.py#L34
     provisioning_profile_path = None
-    if config.get("provisionprofile_dir"):
-        provisioning_profile_path = os.path.join(config["provisionprofile_dir"], provisionprofile_filename)
-        if not os.path.exists(provisioning_profile_path):
+
+    if provisionprofile_filename:
+        pp_dir = config.get("provisioning_profile_dir", None)
+        if not pp_dir:
+            pp_dir = Path(config["work_dir"]).parent / "provisionprofiles"
+            log.warning(f"No provisioning_profile_dir in settings, using default {pp_dir}")
+        provisioning_profile_path = Path(pp_dir) / provisionprofile_filename
+        if not provisioning_profile_path.is_file():
             log.error(f"Could not find provisionprofile file: {provisioning_profile_path}")
             raise IScriptError(f"Could not find provisionprofile file: {provisioning_profile_path}")
-        else:
-            log.info(f"Using provisionprofile {provisioning_profile_path}")
+        log.info(f"Using provisionprofile {provisioning_profile_path}")
 
     await sign_all_apps(config, sign_config, entitlements_path, [app], provisioning_profile_path)
 
     log.info(f"Done signing app {app.app_name}")
 
 
+async def _sign_util(sign_config, binary_path):
+    """Sign command for VPN util binaries.
+
+    Args:
+        sign_config (dict): the task config
+        binary_path (str): the binary name
+    """
+    sign_command = [
+        "codesign",
+        "--timestamp",
+        "-fv",
+        "-s",
+        sign_config["identity"],
+        "--keychain",
+        sign_config["signing_keychain"],
+        "--options",
+        "runtime",
+        binary_path,
+    ]
+    await retry_async(
+        run_command,
+        args=(sign_command,),
+        kwargs={"cwd": os.path.dirname(binary_path), "exception": IScriptError, "output_log_on_exception": True},
+        retry_exceptions=(IScriptError,),
+    )
+
+
+async def _create_pkg_files(config, sign_config, app):
```

Comment:
Sorry I forgot, but I why do we need a separate `create_pkg_files` again? If the `skip_dirs` thing is working, then shouldn't we be able to continue using the main one?

Reply:
Oh, I see you answered this in a previous review iteration... but in that case, how were we using it before? Wouldn't it be better to have a single function that can handle both?

Reply:
I couldn't find an elegant way of making `create_pkg_files` generic and support both use cases.
The original function is tailored for gecko signing and VPN uses a different method of `pkgbuild` and `productbuild`.
Ideally we would split the logic so each external command is a separate function and manipulates the command input as needed based on configs/constants for each app.